service/history/historyEngine.go

```diff
@@ -1289,6 +1289,13 @@ Update_History_Loop:
 		sizeChecker.completedID = completedID
 		sizeChecker.msBuilder = msBuilder
 
+		binChecksum := request.GetBinaryChecksum()
+		if _, ok := domainEntry.GetConfig().BadBinaries.Binaries[binChecksum]; ok {
+			failDecision = true
+			failCause = workflow.DecisionTaskFailedCauseBadBinary
+			failMessage = fmt.Sprintf("binary %v is already marked as bad deployment", binChecksum)
+			goto End_Of_Process_Decision_Loop
```

Comment:
I think it would be better to avoid using goto, otherwise it will become the first goto in cadence..

Reply:
OK